crepecake/src/main/java/com/google/cloud/tools/crepecake/image/Digest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a SHA-256 digest as defined by the Registry HTTP API v2 reference.
+ *
+ * @see <a
+ *     href="https://docs.docker.com/registry/spec/api/#content-digests">https://docs.docker.com/registry/spec/api/#content-digests</a>
+ */
+public class Digest {
+
+  /** Pattern matches a SHA-256 hash - 32 bytes in lowercase hexadecimal. */
+  private static final String HASH_PATTERN = "[a-f0-9]{64}";
+
+  /** Pattern matches a SHA-256 digest - a SHA-256 hash prefixed with "sha256:". */
+  private static final String DIGEST_PATTERN = "sha256:" + HASH_PATTERN;
+
+  private static final Pattern HashPattern = Pattern.compile(HASH_PATTERN);
```

Comment:
naming seems off for `HashPattern`? https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names, depends on the mutability of `Pattern`

Reply:
Renamed it to HASH_PATTERN, and renamed the other *_PATTERN constants to *_REGEX to be more accurate.